Library/OcConsoleLib/TextOutputBuiltin.c

```diff
@@ -993,11 +970,11 @@ OcUseBuiltinTextOutput (
 
   DEBUG ((DEBUG_INFO, "OCC: Using builtin text renderer scale %u mode %u\n", mFontScale, Mode));
 
-  mConsoleMode = Mode;
+  mConsoleMode = EfiConsoleControlScreenText;
   Status       = AsciiTextResetEx (&mAsciiTextOutputProtocol, TRUE, TRUE);
 
   if (!EFI_ERROR (Status)) {
-    OcConsoleControlInstallProtocol (&mConsoleControlProtocol, NULL, NULL);
+    OcConsoleControlInstallProtocol (&mConsoleControlProtocol, &mOriginalConsoleControlProtocol, &mConsoleMode);
     OcConsoleControlSetMode (Mode);
```

Comment:
Prior to b07843fe1d47454747ae4eda9ea0189aa9fb8c03 , OcConsoleControlSetMode came first, which means we were really relying on the native implementation to (truly) set graphics mode, all along.